activation/poet.go

```diff
@@ -443,21 +425,48 @@ func NewPoetServiceWithClient(
 	logger *zap.Logger,
 	opts ...PoetServiceOpt,
 ) *poetService {
-	poetClient := &poetService{
+	service := &poetService{
 		db:                 db,
 		logger:             logger,
 		client:             client,
 		requestTimeout:     cfg.RequestTimeout,
-		certifierInfoCache: cachedData[*certifierInfo]{ttl: cfg.CertifierInfoCacheTTL},
+		certifierInfoCache: cachedData[*types.CertifierInfo]{ttl: cfg.CertifierInfoCacheTTL},
 		powParamsCache:     cachedData[*PoetPowParams]{ttl: cfg.PowParamsCacheTTL},
 		proofMembers:       make(map[string][]types.Hash32, 1),
+		expectedPhaseShift: cfg.PhaseShift,
+		fetchedPhaseShift:  &atomic.Duration{},
 	}
-
 	for _, opt := range opts {
-		opt(poetClient)
+		opt(service)
+	}
+
+	err := service.verifyPhaseShiftConfiguration(context.Background())
+	if err != nil {
+		if errors.Is(err, ErrIncompatiblePhaseShift) {
+			logger.Panic("failed to create poet service",
+				zap.Error(err),
+				zap.String("poet", client.Address()))
+		}
+		logger.Warn("failed to fetch poet phase shift",
+			zap.Error(err),
+			zap.String("poet", client.Address()))
+	}
```

Comment:
NIT: if we already check for a specific error, we don't need to log it any more with `zap.Error(err)`. Only if we do not know what the error is, we add it:

```suggestion
	switch {
	case errors.Is(err, ErrIncompatiblePhaseShift):
		logger.Panic("failed to create poet service: poet has incompatible phase shift",
			zap.String("poet", client.Address()),
		)
	case err != nil:
		logger.Warn("failed to fetch poet phase shift",
			zap.String("poet", client.Address()),
			zap.Error(err),
		)
	}
```